### PR TITLE
[Nano] Add how-to guide for TensorFlow Keras training using bf16

### PR DESCRIPTION
**Pull request by Oscilloscope98**
## Description

Add how-to guide for TensorFlow Keras training using bf16

### How to test?
- [x] Document test: https://yuwentestdocs.readthedocs.io/en/nano-howto-tf-bf16-training/doc/Nano/Howto/Training/TensorFlow/tensorflow_training_bf16.html
- [x] Github preview: https://github.com/Oscilloscope98/BigDL/blob/nano-howto-tf-bf16-training/python/nano/tutorial/notebook/training/tensorflow/tensorflow_training_bf16.ipynb
- [x] Local test on CPX-3

### CODE REVIEW

File: python/nano/tutorial/notebook/training/tensorflow/tensorflow_training_bf16.ipynb [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "> ⚠️ **Warning**\n",
+    ">\n",
+    "> BigDL-Nano will enable intel's oneDNN optimizations by default. ONEDNN BFloat16 are only supported on platforms with AVX512.\n",
+    ">\n",
+    "> Platforms without hardware acceleration for BFloat16 could lead to bad BFloat16 training performance."
```
Comment (Oscilloscope98) [inline python/nano/tutorial/notebook/training/tensorflow/tensorflow_training_bf16.ipynb:88]: I am not sure whether the statement here is correct. Please have a look :)

**Comment by TheaperDeng** [comment]
1. it is recommended to import patch_tensorflow and unpatch_tensorflow from `bigdl.nano.tf`
2. both train_step and test_step should be wrapped with `@nano_bf16`

**Review by plusbang — COMMENTED** [review]
LGTM

**Review by TheaperDeng — APPROVED** [review]
LGTM

**Pull request merged**